openmdao/core/problem.py

```diff
@@ -1013,40 +1172,46 @@ def _assemble_derivative_data(derivative_data, rel_error_tol, abs_error_tol, out
                 else:
-                    # Magnitudes
-                    out_stream.write("  {}: '{}' wrt '{}'\n\n".format(sys_name, of, wrt))
-                    out_stream.write('    Forward Magnitude : {:.6e}\n'.format(magnitude.forward))
-                    out_stream.write('    Reverse Magnitude : {:.6e}\n'.format(magnitude.reverse))
-                    out_stream.write('         Fd Magnitude : {:.6e} ({})\n\n'.format(magnitude.fd,
-                                                                                      fd_desc))
-                    # Absolute Errors
                     error_descs = ('(Jfor  - Jfd) ', '(Jrev  - Jfd) ', '(Jfor  - Jrev)')
-                    for error, desc in zip(abs_err, error_descs):
-                        error_str = _format_error(error, abs_error_tol)
-                        out_stream.write('    Absolute Error {}: {}\n'.format(desc, error_str))
-                    out_stream.write('\n')
-
-                    # Relative Errors
-                    for error, desc in zip(rel_err, error_descs):
-                        error_str = _format_error(error, rel_error_tol)
-                        out_stream.write('    Relative Error {}: {}\n'.format(desc, error_str))
-                    out_stream.write('\n')
-
-                    # Raw Derivatives
-                    out_stream.write('    Raw Forward Derivative (Jfor)\n\n')
-                    out_stream.write(str(forward))
-                    out_stream.write('\n\n')
-
-                    out_stream.write('    Raw Reverse Derivative (Jfor)\n\n')
-                    out_stream.write(str(reverse))
-                    out_stream.write('\n\n')
-
-                    out_stream.write('    Raw FD Derivative (Jfd)\n\n')
-                    out_stream.write(str(fd))
-                    out_stream.write('\n\n')
-
-                    out_stream.write(' -' * 30 + '\n')
+                for error, desc in zip(abs_err, error_descs):
+                    error_str = _format_error(error, abs_error_tol)
+                    logger.info('    Absolute Error {}: {}'.format(desc, error_str))
+                logger.info('')
+
+                # Relative Errors
+                for error, desc in zip(rel_err, error_descs):
+                    error_str = _format_error(error, rel_error_tol)
+                    logger.info('    Relative Error {}: {}'.format(desc, error_str))
+                logger.info('')
+
+                # Raw Derivatives
+                logger.info('    Raw Forward Derivative (Jfor)\n')
+                logger.info(str(forward))
+                logger.info('')
+
+                if not totals:
+                    logger.info('    Raw Reverse Derivative (Jfor)\n')
+                    logger.info(str(reverse))
+                    logger.info('')
+
+                logger.info('    Raw FD Derivative (Jfd)\n')
+                logger.info(str(fd))
+                logger.info('')
+
+                logger.info(' -' * 30)
```

Comment:
I wonder if it would make sense to move some of this code to another file.. since half of the `problem.py` module now seems to consist of "check derivatives" logic, maybe it warrants it's own module similar to `check_config`.